src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -793,6 +793,33 @@ namespace cryptonote
     NOTIFY_NEW_FLUFFY_BLOCK::request fluffy_response;
     fluffy_response.b.block = t_serializable_object_to_blob(b);
     fluffy_response.current_blockchain_height = arg.current_blockchain_height;
+
+    // NOTE: Dupe index check
+    {
+      std::unordered_set<uint64_t> requested_index_set;
+      requested_index_set.reserve(16); // typical maximum number of txs per block
+
+      for (uint64_t requested_index : arg.missing_tx_indices)
+      {
+        if (requested_index_set.find(requested_index) != requested_index_set.end())
```

Comment:
Rather than do a `find` here and `insert` later, it's more efficient (it only has to hash once) to just `insert` and check the return value for a duplicate: the second element of the return pair is whether the insert actually happened:

```C++
if (requested_index_set.insert(requested_index).second)
{
  LOG_ERRO_CCONTEXT
  ...
```

Reply:
Nice, updated. I assume you mean
`if (!requested_index_set.insert(requested_index).second)`
For the bool which indicates insertion success.

Reply:
Whoops, yup.